src/screens/OtherProfile/index.js

```diff
@@ -83,8 +83,9 @@ const OtherProfile = () => {
   const [loadingGenerateAnon, setLoadingGenerateAnon] = React.useState(false);
   const [anonProfile, setAnonProfile] = React.useState();
   const {mappingColorFeed} = useCoreFeed();
-  const {updateFollowDiscoveryContext} = useDiscovery();
-  const [isCurrentFollowed, setIsCurrentFollow] = React.useState(params.data.following);
+  const {updateFollowDiscoveryContext, getIsMeFollowingTargetStatus} = useDiscovery();
+
+  const isCurrentFollowed = getIsMeFollowingTargetStatus(params.data.other_id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change here is a good one as it simplifies the state management by removing the local state `isCurrentFollowed` and replacing it with a function call `getIsMeFollowingTargetStatus(params.data.other_id)`. This reduces the complexity of the component and makes it easier to reason about its behavior.
```diff
- const [isCurrentFollowed, setIsCurrentFollow] = React.useState(params.data.following);
+ const isCurrentFollowed = getIsMeFollowingTargetStatus(params.data.other_id);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->